enhancements/ingress/default-ingress-controller.md

```diff
@@ -0,0 +1,136 @@
+
+End users can now assume that the _only_ way IngressControllers are created are via the installer or through their own API calls.
+
+## Motivation
+
+The operator today has a confusing relationship to the `default` IngressController resource. Not only does the operator confound automation in its competition with users for ownership, but the operator is not even guaranteed to have the context necessary to restore any IngressController that was deleted, including the `default` instance. The operator should have a clear responsibility to manage resources brought by the user — the installer being indistinguishable from a user in this context. The installer should have a clear responsibility to always bring the `default` IngressController resource.
+
+### Goals
+
+1. Eliminate contention with users/automation trying to replace the default IngressController post-installation.
+2. Clarify the role of the ingress-operator with respect to IngressController resource creation
+3. Clarify the role of the installer with respect to the default IngressController resource
+4. Provide a well-known file for end-users to modify the default IngressController prior to installation
+
+### Non-Goals
+
+1. This proposal does not introduce any decision-making into the installer with regards to the shape of the default public IngressController resource. The ingress-operator is still responsible for defaulting the spec field.
+
+## Proposal
+
+The proposal spans two components: the ingress-operator, and the installer.
+
+
+### Ingress Operator
+
+The ingress-operator should be modified to stop creating the `default` IngressController resource.
+
+Because the ingress-operator will no longer ensure a default IngressController resource, status reporting will be enhanced to help alert administrators when the `default` IngressController is missing. The following new ingress operator conditions are proposed:
+
+    MissingDefaultIngressController=[True, False]: True when the `ingresscontrollers/default` resource in the `openshift-ingress-operator` namespace is not found.
+
+When `MissingDefaultIngressController=True`, the operator status should report `Degraded=True` and a `Reason` which cites the `MissingDefaultIngressController` condition.
+
+### Installer
+
+Currently, the installer only renders the `default` IngressController resource when the cluster is configured as "internal". The installer should be changed to _always_ render the `default` IngressController.
+
+The default public IngressController rendered by the installer is the minimum possible resource to make the cluster work. The ingress-operator will handle defaulting to make ingress work out of the box on the platform. The resource the installer will create should look like this:
+
+```
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec: {}
```

Comment:
@abhinavdahiya @Miciah one possible compromise I would like to discuss is setting `.spec.replicas` here. We currently set `.spec.replicas=2` explicitly when the operator creates the resource, because scale clients want the value there. I don't think it's strictly required to make scaling work, but having it there (at least last time I checked) makes for a better UX.

If even setting `.spec.replicas` in the installer is too far, we can try and address the UX implications separately.